### PR TITLE
[windows] build.ps1 polishing

### PR DESCRIPTION
**Pull request by weliveindetail**
Minor modifications to prepare for reduction of complexity in `Build-CMakeProject` function in the the step

### CODE REVIEW

File: utils/build.ps1 [hunk 1]
```diff
@@ -828,6 +828,21 @@ function Get-PinnedToolchainVersion() {
   throw "PinnedVersion must be set"
 }
 
+function Get-ClangDriverName([Platform] $Platform, [string] $Lang) {
```
Comment (compnerd) [inline utils/build.ps1:846]: Should we enumerate the languages? Perhaps that is overly complicating the state.

Comment (weliveindetail) [reply utils/build.ps1:846]: Not sure what you mean, a for-loop with if-else instead of a 2-level switch?

Comment (weliveindetail) [reply utils/build.ps1:846]: Hm, possible

**Comment by weliveindetail** [comment]
Closing this as the above PR contains these patches now.

**Pull request closed without merging**